src/Tribe/Admin_List.php

```diff
@@ -155,6 +158,41 @@ public static function sort_by_event_date( Array $clauses, WP_Query $wp_query )
 			return $clauses;
 		}
 
+		/**
+		 * Defines custom logic for filtering events table by aggregator record.
+		 *
+		 * @param   Array       $clauses    SQL clauses for fetching posts
+		 * @param   WP_Query    $wp_query   A paginated query for items
+		 *
+		 * @return  Array                   Modified SQL clauses
+		 */
+		public static function filter_by_aggregator_record( Array $clauses, WP_Query $wp_query ) {
+			// Check for event post type.
+			if ( $wp_query->get( 'post_type' ) !== Tribe__Events__Main::POSTTYPE ) {
+				return $clauses;
+			}
+
+			// Check if filtering by aggregator record.
+            $record_num = intval( tribe_get_request_var( 'aggregator_record', 0 ) );
+            if( $record_num === 0 ) {
+            	return $clauses;
+            }
+
+			global $wpdb;
+
+            // Add the record meta query if it is missing.
+			if ( ! preg_match( '/tribe_aggregator_record/', $clauses['join'] ) ) {
+				$clauses['join'] .= " LEFT JOIN {$wpdb->postmeta} AS tribe_aggregator_record ON {$wpdb->posts}.ID = tribe_aggregator_record.post_id AND tribe_aggregator_record.meta_key = '_tribe_aggregator_parent_record' ";
+			}
+
+            // Add the record meta filter if it is missing.
```

Comment:
:no_entry_sign: Tabs must be used to indent lines; spaces are not allowed

<sub>hash: 35c102e01344a4558273903faa510f20</sub>